src/hooks/useBalances.ts

```diff
@@ -3,22 +3,52 @@ import isEqual from 'lodash/isEqual'
 import { type SafeBalanceResponse } from '@safe-global/safe-gateway-typescript-sdk'
 import { useAppSelector } from '@/store'
 import { selectBalances } from '@/store/balancesSlice'
+import type { HiddenAssetsOnChain } from '@/store/hiddenAssetsSlice'
+import useHiddenAssets from './useHiddenAssets'
+import { safeFormatUnits, safeParseUnits } from '@/utils/formatters'
+import { BigNumber } from 'ethers'
 
-const useBalances = (): {
+const removeHiddenBalances = (
```

Comment:
`remove` sounds like a destructive operation. `filterHiddenTokens` would be better.